mantidproject/mantid
# Incorrect bounding box calculations for very small cylindrical pixels

**Description of work.**
- [x] new expressions for line intersection with plane
- [x] unit test for bounding box calculation for a capped cylinder detector
- [x] amended [release notes](https://github.com/mantidproject/mantid/blob/9126b057c92cc120344250d81995860c7e2d015b/docs/source/release/v4.3.0/framework.rst#id3)

**To test:**
Upload and unzip the [test IDF](https://github.com/mantidproject/mantid/files/3934612/test_instrument.xml.zip), then run the following script in MantidPlot or the Workbench:
```
w = LoadEmptyInstrument(Filename='/tmp/test_instrument.xml', OutputWorkspace='test')
print(w.componentInfo().shape(42).getBoundingBox().width())
```
Results should be `[0.0055, 0.0042, 0.0055]`

Fixes #27552

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
